doc/cmake.md

```diff
@@ -0,0 +1,47 @@
+# Building a new data model library with CMake
+
+PODIO exposes multiple CMake macros to enable the building of data model libraries.
+For a package developer tutorial on CMake, please have a look [here])https://hsf-training.github.io/hsf-training-cmake-webpage/). PODIO follows the same conventions used therein.
+
+## Exported CMake functions and macros
+The file [cmake/podioMacros.cmake](https://github.com/AIDASoft/podio/blob/master/cmake/podioMacros.cmake) provides the following functions for external use:
+
+  1. PODIO_GENERATE_DATAMODEL - generate the data model from provided yaml file
+  2. PODIO_ADD_DATAMODEL_CORE_LIB - compile the generated data model
+  3. PODIO_ADD_ROOT_IO_DICT - configure ROOT I/O backend and interface to Python
+  4. PODIO_ADD_SIO_IO_BLOCKS - configure SIO backend
+
+Their exact parameters and functionality are documented in the file mentioned above. Below a few examples.
+
+## A simple example
+After a proper `find_package(PODIO)`, the C++ code of the new data model `NewDM` can be created and compiled via
```

Comment:
```suggestion
After a proper `find_package(PODIO)`, the C++ code of the new data model `newdm` can be created and compiled via
```
Just for consistency with the snippets below (or alternatively change the ones below).